### ISSUE TITLE
Method could be a function

### ISSUE THREAD

**Issue by 1Mark** [PyCQA/pylint#7106]
### Bug description

Seems like there is a regression between 2.14.4 and 2.8.2 since in 2.8.2 it flags `[R0201(no-self-use), Blah.get] Method could be a function` but in 2.14.4 it doesn't

### Command used

```shell
pylint folder/file.py
```

### Pylint output

```shell
10/10
```

### Expected behavior

[R0201(no-self-use), Blah.get] Method could be a function

### Pylint version

```shell
2.14.4
```

**Comment by DanielNoord** [comment]
Please see the changelog of `2.14`:
https://pylint.pycqa.org/en/latest/whatsnew/2/2.14/summary.html

We have made `no-self-use` an optional checker. You'll need to turn it on in your configuration if you want to keep checking for that message!
